Added functionality to add the candidates while creating election  #73

rairohit17 opened the pull request:
resolves #73 
. changed the createElection functions parameters to pass  a candidates array  upon initialization
. created suitable frontend changes to allow users to add , remove ,update candidates while creating an election 
. updated tests so that they do not fail due to change in contract 's function

https://github.com/user-attachments/assets/f7f915aa-9044-4518-ba48-8340fcf5b4bb

. made sure all tests pass 


https://github.com/user-attachments/assets/0e23a309-100f-4ca0-ae91-ee5c9996b988

rairohit17 commented:
@Ronnieraj37  could you review the pr and tell if any changes are required
i have yet not not added tests for the updated function , just updated the existing ones  , we can add some tests for the updated functions as well

Ronnieraj37 commented:
Hey @rairohit17 Pl add a check for _candidates .length >= 2 in the createElection from the solidity side. So that we do not have bots creating empty elections from block explorer.

rairohit17 commented:
hey @Ronnieraj37  i have added a check for _candidates.length>2 also i have added a warning for the same in frontend 

Apart from that i have updated all tests  and made them compatible with the contract's changes 
i have also added optimizer for reducing the contract's bytecode size and lower gas costs when deploying as the contracts byecode exceeded the limit 
please review and let me know if any changes are required

Ronnieraj37 commented:
Hey @rairohit17 the changes look good. Just fix the merge conflicts!

rairohit17 commented:
@Ronnieraj37  resolved all merge conflicts
